quickcheck-dynamic/src/Test/QuickCheck/DynamicLogic/Quantify.hs

```diff
@@ -51,10 +63,9 @@ generateQ q = fromJust (genQ q) `suchThat` isaQ q
 shrinkQ :: Quantification a -> a -> [a]
 shrinkQ q a = filter (isaQ q) (shrQ q a)
 
--- | Wrap a `Gen a` generator in a `Quantification a`.
--- Uses given shrinker.
-withGenQ :: Gen a -> (a -> [a]) -> Quantification a
-withGenQ gen = Quantification (Just gen) (const True)
+-- | Construct a `Quantification a` from its constituents.
```

Comment:
I would like to see some better documentation here explaining that it is important for the `(a -> Bool)` to be sound and complete w.r.t. the generator.

Reply:
I think this is documented in the `Quantification` haddocks

Reply:
But that means the user has to read that documentation to use this function correctly - isn't that a shame?

Reply:
When I think about it, one way to make sure it only has to be sound (in the sense of only being True on things that could have been generated) is to use `suchThat` on the Gen. That way the user doesn't have to ensure that it is True on _all_ things that were generated, that will be the case automatically.